src/org/thoughtcrime/securesms/loki/todo/AvatarPlaceholderGenerator.kt

```diff
@@ -0,0 +1,80 @@
+package org.thoughtcrime.securesms.loki.todo
+
+import android.content.Context
+import android.graphics.*
+import android.graphics.drawable.BitmapDrawable
+import android.text.TextPaint
+import android.text.TextUtils
+import androidx.annotation.ColorInt
+import androidx.core.graphics.ColorUtils
+import network.loki.messenger.R
+import java.util.*
+
+object AvatarPlaceholderGenerator {
+
+    private const val EMPTY_LABEL = "0";
+
+    private val tmpFloatArray = FloatArray(3)
+
+    fun generate(context: Context, pixelSize: Int, hashString: String, displayName: String?): BitmapDrawable {
+        //TODO That should be replaced with a proper hash extraction code.
+        val hash: Long
+        if (hashString.length >= 12 && hashString.matches(Regex("^[0-9A-Fa-f]+\$"))) {
+            hash = hashString.substring(0 until 12).toLong(16)
+        } else {
+            hash = 0
+        }
+
+        // Do not cache color array, it may be different depends on the current theme.
+        val colorArray = context.resources.getIntArray(R.array.user_pic_placeholder_primary)
+        val colorPrimary = colorArray[(hash % colorArray.size).toInt()]
+        val colorSecondary = changeColorHueBy(colorPrimary, 12f)
+
+        val labelText = when {
+            !TextUtils.isEmpty(displayName) -> extractLabel(displayName!!)
+            !TextUtils.isEmpty(hashString) -> extractLabel(hashString)
+            else -> EMPTY_LABEL
+        }
+
+        val bitmap = Bitmap.createBitmap(pixelSize, pixelSize, Bitmap.Config.ARGB_8888)
+        val canvas = Canvas(bitmap)
+
+        // Draw background/frame
+        val paint = Paint(Paint.ANTI_ALIAS_FLAG)
+        paint.shader = LinearGradient(0f, 0f, 0f, pixelSize.toFloat(),
```

Comment:
Again, let's first get the design going cross-platform before changing it :)

Reply:
Easy, this fancy gradient doesn't change that much anyway :)